src/main/java/io/gravitee/policy/json2xml/JsonToXmlTransformationPolicy.java

```diff
@@ -40,62 +41,81 @@
  */
 public class JsonToXmlTransformationPolicy extends JsonToXmlTransformationPolicyV3 implements Policy {
 
+    private static final String MESSAGE_INVALID_PAYLOAD_FAILURE_KEY = "JSON_INVALID_MESSAGE_PAYLOAD";
+
     public JsonToXmlTransformationPolicy(final JsonToXmlTransformationPolicyConfiguration configuration) {
         super(configuration);
     }
 
+    private static void applyHeaders(HttpHeaders headers, Buffer xmlBuffer) {
```

Comment:
Maybe my suggestion is not the best but `applyHeaders` is to generic in my opinion
```suggestion
    private static void setContentHeaders(HttpHeaders headers, Buffer xmlBuffer) {
```